http://sources.redhat.com/cgi-bin/cvsweb.cgi/LVM2/udev/69-dm-lvm-metad.rules.diff?r1=1.3&r2=1.4&cvsroot=lvm2

X-Gentoo-Bug: 409921
X-Gentoo-Bug-URL: https://bugs.gentoo.org/show_bug.cgi?id=409921

===================================================================
RCS file: /cvs/lvm2/LVM2/udev/69-dm-lvm-metad.rules,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- LVM2/udev/69-dm-lvm-metad.rules	2012/03/06 02:30:50	1.3
+++ LVM2/udev/69-dm-lvm-metad.rules	2012/03/12 08:59:55	1.4
@@ -19,7 +19,12 @@
 # Device-mapper devices are processed only on change event or on supported synthesized event.
 KERNEL=="dm-[0-9]*", ENV{DM_UDEV_RULES_VSN}!="?*", GOTO="lvm_end"
 
+# Set proper sbin path, /sbin has higher priority than /usr/sbin.
+ENV{LVM_SBIN_PATH}="/sbin"
+TEST!="$env{LVM_SBIN_PATH}/lvm", ENV{LVM_SBIN_PATH}="/usr/sbin"
+TEST!="$env{LVM_SBIN_PATH}/lvm", GOTO="lvm_end"
+
 # Only process devices already marked as a PV - this requires blkid to be called before.
-ENV{ID_FS_TYPE}=="LVM2_member|LVM1_member", RUN+="$env{DM_SBIN_PATH}/pvscan --cache --major $major --minor $minor"
+ENV{ID_FS_TYPE}=="LVM2_member|LVM1_member", RUN+="$env{LVM_SBIN_PATH}/lvm pvscan --cache --major $major --minor $minor"
 
 LABEL="lvm_end"
